### PR TITLE
fix(dynamite): Allow int enums

### LINKED ISSUE #1240: Integer enums are broken
A schema like
```json
"enum-int": {
    "type": "integer",
    "enum": [
        0,
        1,
        2
    ]
}
```
does not work because we expect the possible enum values to be only strings.

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes #1240 
replaces: #1242

### CODE REVIEW

File: packages/dynamite/dynamite/lib/src/models/type_result/enum.dart [hunk 1]
```diff
@@ -20,7 +20,7 @@ class TypeResultEnum extends TypeResult {
     final bool onlyChildren = false,
     final String? mimeType,
   }) {
-    if (subType.name == 'String') {
+    if (subType.name == 'String' || subType.name == 'int') {
       return '$object.name';
     }
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:25]: Shouldn't this return the name in all cases regardless of the subtype? I guess that was left over from the changes I did

**Review by provokateurin — CHANGES_REQUESTED** [review]
Great job!

File: packages/dynamite/dynamite/lib/src/models/type_result/enum.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:25]: I'd rather merge #1323 as it will remove this mess altogether :)

Comment (provokateurin) [reply packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:25]: Ok let's get that merged first then

**Comment by Leptopoda** [comment]
I'm not sure this is the best way to do it.
I think that enum_value.name should always return a `String`.
What about introducing a separate attribute called `value` that is of type `T`?

**Comment by provokateurin** [comment]
I like the idea, please go ahead

**Comment by provokateurin** [comment]
I rebased the branch and confirmed it works with the latest specs. If you want I can push that branch if you haven't done it locally already.

**Comment by Leptopoda** [comment]
I've already rebased and done this:
>  What about introducing a separate attribute called value that is of type T?

I'll push it in a few minutes just want to make sure it can be easily adapted without any breaking changes.

**Comment by Leptopoda** [comment]
I left it on a separate commit for now so we can roll it back easier.
We can squash them before merging.

**Comment by provokateurin** [comment]
I'm confused what you did in the second commit. It looks like you reverted a lot of the changes from the first commit?

**Comment by Leptopoda** [comment]
You are right.
In the second commit we just have the value getter and zan use the built_value_generator for the rest of the enum so no need to do it ourselves.

File: packages/dynamite/dynamite_end_to_end_test/test/enum_test.dart [hunk 2]
```diff
@@ -3,23 +3,96 @@ import 'package:dynamite_end_to_end_test/enum.openapi.dart';
...
+          result.value,
+          specifiedType: const FullType(EnumDynamic),
+        );
+        expect(deserialized, result.key);
+      }
+    });
+  });
+
+  group('wrapped', () {
+    test('serialize', () {
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/enum_test.dart:82]: Deserialize test for the wrapped would be good too.

**Review by provokateurin — APPROVED** [review]

**Pull request merged**